norduser/service/fork.go

```diff
@@ -3,63 +3,109 @@ package service
 import (
 	"errors"
 	"fmt"
+	"log"
 	"os/exec"
+	"strconv"
 	"strings"
+	"sync"
 	"syscall"
 
 	"github.com/NordSecurity/nordvpn-linux/internal"
-	"golang.org/x/sys/unix"
 )
 
 // ErrNotStarted when disabling norduser
 var ErrNotStarted = errors.New("norduserd wasn't started")
 
 // ChildProcessNorduser manages norduser service through exec.Command
 type ChildProcessNorduser struct {
-	commandHandles map[uint32]*exec.Cmd
+	mu sync.Mutex
 }
 
 func NewChildProcessNorduser() *ChildProcessNorduser {
-	return &ChildProcessNorduser{
-		commandHandles: make(map[uint32]*exec.Cmd),
+	return &ChildProcessNorduser{}
+}
+
+// handlePsError returns nil if err is nil or error code 1(no processes listed). It returns unmodified err in any other
+// case.
+func handlePsError(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	var exiterr *exec.ExitError
+	if errors.As(err, &exiterr) {
+		// ps returns 1 when no processes are shown
+		if exiterr.ExitCode() == 1 {
```

Comment:
from what I've saw `ps` is always returning `1` for error, maybe this needs to be combined with the output if it is empty, otherwise all the errors might be ignored?